addons/mass_mailing/models/mass_mailing.py

```diff
@@ -612,7 +612,7 @@ def _group_expand_states(self, states, domain, order):
     def update_opt_out(self, email, list_ids, value):
         if len(list_ids) > 0:
             model = self.env['mail.mass_mailing.contact'].with_context(active_test=False)
-            records = model.search([('email', '=ilike', email)])
+            records = model.search([('email_normalized', '=', email)])
```

Comment:
As this is a public method you should not make any assumptions on email and normalize it before managing it.